casper/src/main/scala/coop/rchain/casper/TuplespaceEvent.scala

```diff
@@ -20,31 +20,38 @@ case object Receive extends Polarity
 trait Cardinality
 case object Linear    extends Cardinality
 case object NonLinear extends Cardinality
+case object Peek      extends Cardinality
 
 object TuplespaceEvent {
 
   private[this] def toOperation(produce: Produce): TuplespaceOperation =
     TuplespaceOperation(Send, if (produce.persistent) NonLinear else Linear, produce.hash)
 
-  private[this] def toOperation(consume: Consume): TuplespaceOperation =
-    TuplespaceOperation(Receive, if (consume.persistent) NonLinear else Linear, consume.hash)
+  private[this] def toOperation(consume: Consume, peek: Boolean): TuplespaceOperation =
+    TuplespaceOperation(
+      Receive,
+      if (consume.persistent) NonLinear else if (peek) Peek else Linear,
+      consume.hash
+    )
 
   def from(produce: Produce): (Blake2b256Hash, TuplespaceEvent) =
     produce.channelsHash -> TuplespaceEvent(toOperation(produce), None)
 
   def from(consume: Consume): Option[(Blake2b256Hash, TuplespaceEvent)] = consume match {
     case Consume(singleChannelHash :: Nil, _, _, _) =>
-      Some(singleChannelHash -> TuplespaceEvent(toOperation(consume), None))
+      Some(singleChannelHash -> TuplespaceEvent(toOperation(consume, false), None))
```

Comment:
This is only loosely related to this PR

Reply:
Consume has this flag on the tuplespace level, it doesn't in the replay log because this flag is only relevant in replay when a COMM event happened. Since changing this is not "a must" for the peek mechanism to work, let's consider this idea and it's impact in a separate ticket

Reply:
It's also relevant for saying whether two event logs are conflicting. We're essentially abusing the fact that PX has the same merging rules as 4X, without expressing it in code. Let's then have a separate PR addressing this.